app/printing/printFull/get.controller.js

```diff
@@ -1,13 +1,75 @@
 const { logger } = require('../../../common/logging/logger')
-const { removeUrlLevels } = require('../../../common/utils/util')
+const { removeUrlLevels, groupBy, getYearMonthFromDate, getStatusText } = require('../../../common/utils/util')
+const { getCommentText } = require('../../../common/utils/getCommentText')
+const { getSentencePlan } = require('../../../common/data/sentencePlanningApi')
+const { COMMENT_TYPES, ACTION_STATUS_TYPES } = require('../../../common/utils/constants')
 
-const printFullSentencePlan = async ({ path, params: { id, planId } }, res) => {
+const printFullSentencePlan = async ({ path, params: { id, planId }, tokens }, res) => {
   try {
     const backUrl = removeUrlLevels(path, 1)
-    res.render(`${__dirname}/index`, { backUrl })
+
+    let objectives
+    let comments
+
+    // get the sentence plan
+    try {
+      ;({ comments, objectives = [] } = await getSentencePlan(planId, tokens))
+    } catch (error) {
+      logger.error(`Could not retrieve active sentence plan ${planId} for offender ${id}, error: ${error}`)
+      return res.render('app/error', { error })
+    }
+
+    // determine the status of each objective
+    objectives = objectives.map(objective => {
+      const currentObjective = objective
+
+      // objectives default to active if not caught with the rules below
+      currentObjective.type = 'active'
+
+      if (currentObjective.actions.every(({ status }) => [ACTION_STATUS_TYPES.NOT_STARTED].includes(status))) {
+        currentObjective.type = 'future'
+      } else if (
+        currentObjective.actions.every(({ status }) =>
+          [
+            ACTION_STATUS_TYPES.COMPLETED,
+            ACTION_STATUS_TYPES.PARTIALLY_COMPLETED,
+            ACTION_STATUS_TYPES.ABANDONED,
+          ].includes(status)
```

Comment:
I think we identify a completed status in multiple places actually. I think this is worth creating a utility function. The plan view page certainly needs to do it 🤔